epytope/EpitopePrediction/External.py

```diff
@@ -3937,15 +3967,21 @@ def parse_external_result(self, file):
         :return: A dictionary containing the prediction results
         :rtype: dict
         """
-        result = defaultdict(defaultdict)
         f = csv.reader(open(file, "r"), delimiter='\t')
-        alleles = [x.replace("*", "_").replace(":", "") for x in set([x for x in next(f) if x != ""])]
+        scores = defaultdict(defaultdict)
+        ranks = defaultdict(defaultdict)
+        # NetMHCiipan 3.1 inherits from NetMHCiipan 3.0 thus has a slighty different allele nomenclature
+        alleles = [Allele('HLA-'+x) if '*' in x else Allele('HLA-'+x[:7].replace("_","*")+':'+x[7:]) for x in set([x for x in next(f) if x != ""])]
```

Comment:
do you have an example how the expected content will be formatted?

Reply:
One of the tools has allele format `A*02:01`, the other one `A_0201`. It will be formatted to `HLA-A*02:01` such that a `Allele Obj` can be constructed based on this. It's a very ugly way of doing it, but since one class gets inherited and the input is not 100% identical, I needed to do this distinction. Also `set` has here been used to remove duplicated alleles, which happens in that tools for some reason.

Reply:
Not sure if I understand your command correctly but I think you are referring to the fact that the formats are different and it does not work for a specific NetMHCpan? version? You can still override the respective function even if you inherit from another class. I guess that would solve the issue here?

Reply:
That would solve the issue yes 👍 I just thought overriding it is less preferred than a simple if here. But will change that